PayPalCheckout/src/main/java/com/paypal/android/checkout/paymentbutton/PayPalButton.kt

```diff
@@ -0,0 +1,232 @@
+        }
+    }
+}
+
+/**
+ * Defines the colors available for PayPal buttons.
+ *
+ * @see GOLD is the default color if one is not provided and is the recommended choice as research
+ * has shown it results in the best conversion.
+ * @see BLUE is the preferred alternative color if gold does not work for your experience. Research
+ * has shown that people know it is our brand color, which provides a halo of trust and security to
+ * your experience.
+ * @see WHITE is one of our secondary alternatives. This color is less capable of drawing people's
+ * attention.
+ * @see BLACK is one of our secondary alternatives. This color is less capable of drawing people's
+ * attention.
+ * @see SILVER is one of our secondary alternatives. This color is less capable of drawing people's
+ * attention.
+ */
+enum class PayPalButtonColor(
+    val value: Int,
+    override val colorResId: Int,
+    override val hasOutline: Boolean = false,
+    override val luminance: PaymentButtonColorLuminance
+) : PaymentButtonColor {
+    GOLD(value = 0, colorResId = R.color.paypal_gold, luminance = PaymentButtonColorLuminance.LIGHT),
+    BLUE(value = 1, colorResId = R.color.paypal_blue, luminance = PaymentButtonColorLuminance.DARK),
+    WHITE(
+        value = 2,
+        colorResId = R.color.paypal_white,
+        hasOutline = true,
+        luminance = PaymentButtonColorLuminance.LIGHT
+    ),
+    BLACK(value = 3, colorResId = R.color.paypal_black, luminance = PaymentButtonColorLuminance.DARK),
+    SILVER(value = 4, colorResId = R.color.paypal_silver, luminance = PaymentButtonColorLuminance.LIGHT);
+
+    companion object {
+        /**
+         * Given an [attributeIndex] this will provide the correct [PayPalButtonColor]. If an
+         * invalid [attributeIndex] is provided then it will throw an [IllegalStateException].
+         *
+         * @throws [IllegalArgumentException] when an invalid index is provided.
+         */
+        operator fun invoke(attributeIndex: Int): PayPalButtonColor {
+            return when (attributeIndex) {
+                GOLD.value -> GOLD
+                BLUE.value -> BLUE
+                WHITE.value -> WHITE
+                BLACK.value -> BLACK
+                SILVER.value -> SILVER
+                else -> throw createFormattedIllegalArgumentException("PayPalButtonColor", values().size)
+            }
+        }
+    }
+}
+
+/**
+ * Defines the labels available for payment buttons. If no label is provided then it will
+ * default to [PAYPAL] which will default to no label and only displaying the PayPal wordmark. For
+ * other labels they will have the label value itself along with a position either at the start or
+ * the end of the button.
+ *
+ * @see PAYPAL this is the default label which does not display a label at all.
+ * PayPalButton.
+ */
+enum class PayPalButtonLabel(
+    val value: Int,
+    val position: Position? = null,
+    private val stringResId: Int? = null
+) {
+    PAYPAL(value = 0),
+    CHECKOUT(
```

Comment:
Should each of these labels have a doc string telling merchants when the particular label should be used?

Reply:
I think a small kdoc outlining each of the label options will be helpful.

Reply:
@sarahkoop @tdchow I added some doc, let me know if that's enough or if I should include more information